pkg-manager/core/test/catalogs.ts

```diff
@@ -379,6 +380,61 @@ test('lockfile catalog snapshots do not contain stale references on --filter', a
   expect(loadJsonFile<ProjectManifest>(pathToIsPositivePkgJson)?.version).toBe('3.1.0')
 })
 
+// Regression test for https://github.com/pnpm/pnpm/issues/8639
+test('--fix-lockfile with --filter does not erase catalog snapshots', async () => {
```

Comment:
I double checked that this test fails before the fix here.

<img width="842" alt="Screenshot 2025-02-23 at 3 42 49 PM" src="https://github.com/user-attachments/assets/189349ee-78a7-445e-a9ba-9904dc0809a6" />